catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Segment.view.js

```diff
@@ -0,0 +1,183 @@
+                    data = this.model.toJSON();
+                }
+                data.editableSegment = this.editableSegment;
+                if (!this.model.get('multiValued') && this.model.constructTitle) {
+                    data.title = this.model.constructTitle();
+                } else {
+                    data.title = this.model.get('segmentName');
+                }
+                data.customizable = FieldDescriptors.isCustomizableSegment(this.model.get('segmentType')) && !this.model.get('multiValued');
+                data.showFields = this.model.get('fields').models.length > 0;
+                data.showAssociations = FieldDescriptors.isCustomizableSegment(this.model.get('segmentType')) && this.model.get('fields').models.length > 0;
+                return data;
+            },
+            /**
+             * Set up the popovers based on if the selector has a description.
+             */
+            setupPopOvers: function () {
+                var view = this;
+                this.model.get('fields').models.forEach(function (each) {
+                    if (!_.isUndefined(each.get("desc"))) {
+                        var options,
+                            selector = ".description[data-title='" + each.get('key') + "']";
+                        options = {
+                            title: each.get("name"),
+                            content: each.get("desc"),
+                            trigger: 'hover'
+                        };
+                        view.$(selector).popover(options);
+                    }
+                });
+            }
+        });
+
+        Segment.SegmentCollectionView = Marionette.CompositeView.extend({
+            template: 'segmentList',
+            itemView: Segment.SegmentView,
+            events: {
+                "click .add-segment": 'addSegment'
+            },
+            initialize: function () {
+                this.listenTo(wreqr.vent, 'removeSegment:' + this.model.get('segmentId'), this.removeSegment);
+            },
+            buildItemView: function (item, ItemViewType, itemViewOptions) {
+                var options = _.extend({
+                    model: item,
+                    editableSegment: this.model.get('multiValued')
+                }, itemViewOptions);
+                return new ItemViewType(options);
+            },
+            addSegment: function () {
+                var selector = this.$('.auto-populate-selector');
+                var prePopulateId;
+                if (selector) {
+                    var selected = selector.find(':selected');
+                    if (selected.attr('id') !== 'empty') {
+                        prePopulateId = selected.attr('id');
+                    }
+                }
+                this.model.addSegment(prePopulateId);
+                this.render();
+            },
+            removeSegment: function (id) {
+                this.model.removeSegment(id);
+                this.render();
```

Comment:
You could just listen to the model and the view will call render when the field you care about changes.